### PR TITLE
Add Invoices Tab to a Subscription View

### PR DESCRIPTION
**Pull request by sravfeyn**
See http://manage.dimagi.com/default.asp?163274

screen shot 2015-04-08 at 7 27 56 pm

### CODE REVIEW

File: corehq/apps/accounting/views.py [hunk 1]
```diff
@@ -338,8 +338,27 @@ def cancel_form(self):
...
+    def invoice_context(self):
+        subscriber_domain = self.subscription.subscriber.domain
+
+        invoice_report = InvoiceInterface(self.request)
+        invoice_report.filters.update(subscription__subscriber__domain=subscriber_domain)
+
+        invoice_report_url = invoice_report.get_url()
+        invoice_export_url = invoice_report.get_url(render_as='export')
+        # Tied to InvoiceInterface. Not a neat way of constructing URL with GET params
+        get_params_string = "?subscriber={domain_name}".format(domain_name=subscriber_domain)
```
Comment (esoergel) [inline corehq/apps/accounting/views.py]: You should be able to do this:
```python
import urllib
get_params_string = "?" + urllib.urlencode({'subscriber': subscriber_domain})
```
That'll escape stuff as needed to put in a URL, although that's probably not an issue with the restrictions we put on domain names

Comment (esoergel) [inline corehq/apps/accounting/views.py]: or maybe
```python
from urllib import urlencode
url_params = {'subscriber': subscriber_domain}
invoice_report_url = "{}?{}".format(invoice_report.get_url(), urlencode(params))
invoice_export_url = "{}?{}".format(invoice_report.get_url(render_as='export'), urlencode(params))
```
I dunno, something along those lines

**Comment by esoergel** [comment]
looks good to me aside from the urlencoding thing, but that's not blocking, so I'll merge.

File: corehq/apps/accounting/views.py [hunk 2]
```diff
@@ -338,8 +338,27 @@ def cancel_form(self):
         return CancelForm()
 
     @property
-    def page_context(self):
+    def invoice_context(self):
+        subscriber_domain = self.subscription.subscriber.domain
+
+        invoice_report = InvoiceInterface(self.request)
+        invoice_report.filters.update(subscription__subscriber__domain=subscriber_domain)
```
Comment (nickpell) [inline corehq/apps/accounting/views.py]: This is not the same as invoices for the subscription - a domain can have different subscriptions at different points in time, so this query can return invoices not from this subscription.  We should fix that, or if that's the intended use case, clarify that in the UI.

File: corehq/apps/accounting/views.py (hunk 1 above)
Comment (nickpell) [inline corehq/apps/accounting/views.py]: see other comment

File: corehq/apps/accounting/views.py [hunk 3]
```diff
@@ -338,8 +338,27 @@ def cancel_form(self):
...
+
+        invoice_report_url = invoice_report.get_url()
+        invoice_export_url = invoice_report.get_url(render_as='export')
+        # Tied to InvoiceInterface. Not a neat way of constructing URL with GET params
```
Comment (nickpell) [inline corehq/apps/accounting/views.py]: see other comment about the query

File: corehq/apps/accounting/views.py (hunk 2 above)
Comment (sravfeyn) [inline corehq/apps/accounting/views.py]: This came out in the scoping I did with Krishna. He said, billing-team usually want to see all the Invoices for a project space, not particularly for the selected subscription. Should we still change this, if so good to ping the fogbugz case http://manage.dimagi.com/default.asp?163274?

Comment (nickpell) [inline corehq/apps/accounting/views.py]: thanks, responded in the case

**Pull request merged**